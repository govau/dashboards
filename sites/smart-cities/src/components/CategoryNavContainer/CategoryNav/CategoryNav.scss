@import '~style/main';

.wrapper {
  position: relative;
  width: 100%;
  height: $top-nav-height-mobile;
  background: $color-white;
  z-index: $z-index-cat-nav;

  // Colors are all inverted on the overview page
  &.wrapper__overview {
    background: transparent;
  }

  @include for-desktop-up {
    height: $top-nav-height-desktop;
    padding-left: 0;
  }
}

.categoryNav {
  padding-left: 16px;
  display: flex;
  white-space: nowrap;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
  -ms-overflow-style: -ms-autohiding-scrollbar;

  @include for-desktop-up {
    @include container;
  }
}

.link {
  height: 100%;
  display: inline-block;
  padding: 8px 0;
  margin-right: 10px;
  text-decoration: none;

  @include for-desktop-up {
    margin-right: 8px;
    padding: 20px 0;
  }

  @include for-big-desktop-up {
    margin-right: 10px;
  }

  &:last-child {
    padding-right: 16px;
  }

  &:hover,
  &.link__active {
    .linkText {
      color: $color-white;
    }

    // overview and context have the same style for hover and active
    // the other links are different for hover and active
    .overview {
      border: 2px solid $color-grey--500;
      color: $color-grey--700;

      .wrapper__overview & {
        color: $color-white;
        border-color: $color-primary--500;
      }
    }

    .primary {
      border: 2px solid $color-primary--600;
      color: $color-grey--700;

      .wrapper__overview & {
        color: $color-white;
        border-color: $color-white;
      }
    }
  }

  &:hover {
    .linkText {
      @include shadow;

      border-color: transparent;
    }

    .jobs {
      background: $color-jobs--500;
    }

    .infrastructure {
      background: $color-infrastructure--500;
    }

    .housing {
      background: $color-housing--500;
    }

    .liveability {
      background: $color-liveability--500;
    }

    .innovation {
      background: $color-innovation--500;
    }

    .planning {
      background: $color-planning--500;
    }
  }

  &.link__active {
    .jobs {
      background: $color-jobs--600;
    }

    .infrastructure {
      background: $color-infrastructure--600;
    }

    .housing {
      background: $color-housing--900; // 900 to pass WCAG AA
    }

    .liveability {
      background: $color-liveability--600;
    }

    .innovation {
      background: $color-innovation--600;
    }

    .planning {
      background: $color-planning--600;
    }
  }
}

.linkText {
  @include shadow;

  padding: 0 24px;
  border-radius: 100px;
  color: $color-grey--700;
  border: 2px solid transparent;
  font-size: 14px;
  transition: $animation-duration; // box-shadow and color

  .wrapper__overview & {
    color: $color-white;
    border-color: $color-white-20;
    box-shadow: none;
  }

  @include for-desktop-up {
    padding: 0 12px;
    font-size: 16px;
  }

  @include for-big-desktop-up {
    padding: 0 24px;
  }
}
