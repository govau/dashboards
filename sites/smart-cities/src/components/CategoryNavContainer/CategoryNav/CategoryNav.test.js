import React from 'react';
import { shallow } from 'enzyme';
import CategoryNav from './CategoryNav';
import {
  CATEGORY_IDS,
  NO_CITY,
} from '../../../constants';

jest.mock('../../Pill/Pill', () => 'Pill');
jest.mock('react-router-dom', () => ({
  NavLink: jest.fn(),
}));

const mockElement = {
  offsetLeft: 200,
  parentElement: {
    scrollLeft: 0,
  },
};

// mock this out so that componentDidMount returns the mock element
document.querySelector = jest.fn(() => mockElement);

const defaultProps = {
  cityId: 'perth',
  categoryId: CATEGORY_IDS.OVERVIEW,
  categories: [
    {
      name: 'Perth',
      id: 'perth',
    },
    {
      name: 'Sydney',
      id: 'sydney',
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <CategoryNav {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should add a modifier class on the all cities overview page', () => {
  const component = shallow(
    <CategoryNav
      {...defaultProps}
      cityId={NO_CITY}
    />
  );

  expect(component.find('.wrapper').hasClass('wrapper__overview')).toBe(true);
});

it('should not add a modifier class on a city overview page', () => {
  const component = shallow(
    <CategoryNav
      {...defaultProps}
      city="some city"
    />
  );

  expect(component.find('.wrapper').hasClass('wrapper__overview')).toBe(false);
});

it('should scroll the nav item into view when clicked', () => {
  const component = shallow(
    <CategoryNav {...defaultProps} />
  );

  const eventMock = {
    target: {
      parentElement: mockElement,
    },
  };

  component.find('.link').at(2).simulate('click', eventMock);

  expect(mockElement.parentElement.scrollLeft).toBe(mockElement.offsetLeft - 40);
});
