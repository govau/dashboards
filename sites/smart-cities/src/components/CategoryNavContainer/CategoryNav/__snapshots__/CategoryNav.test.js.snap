// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\">
  <div className=\\"categoryNav\\">
    <mockConstructor className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/perth/overview\\" id=\\"overview-link\\" aria-label=\\"Overview of All cities\\">
      <Pill className=\\"linkText overview\\" height={28}>
        Overview
      </Pill>
    </mockConstructor>
    <mockConstructor className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/perth/perth\\" onClick={[Function]} aria-label=\\"Perth in All cities\\">
      <Pill className=\\"linkText undefined\\" height={28}>
        Perth
      </Pill>
    </mockConstructor>
    <mockConstructor className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/perth/sydney\\" onClick={[Function]} aria-label=\\"Sydney in All cities\\">
      <Pill className=\\"linkText undefined\\" height={28}>
        Sydney
      </Pill>
    </mockConstructor>
  </div>
</div>"
`;
